src/components/OrgFile/components/AgendaModal/components/AgendaDay/index.js

```diff
@@ -42,18 +50,32 @@ export default class AgendaDay extends PureComponent {
           }
 
           if (planningItem.get('type') === 'DEADLINE') {
-            if (planningItemDate < moment() && isToday) {
-              return true;
-            }
-
-            if (timestamp.get('delayType') === '-' && isToday) {
-              const delayUnit = momentUnitForTimestampUnit(timestamp.get('delayUnit'));
-              const appearDate = planningItemDate
-                .clone()
-                .subtract(timestamp.get('delayValue'), delayUnit);
-
-              return date >= appearDate;
+            //---------------------------
+            // HANDLING THE CURRENT DAY
+            if (isToday) {
+              // PAST DEADLINES (ALWAYS SHOW)
+              if (planningItemDate < moment()) {
+                return true;
+              }
+
+              if (timestamp.get('delayType') === '-') {
+                // DELAY IS EXPLICITY SET
+                const delayUnit = momentUnitForTimestampUnit(timestamp.get('delayUnit'));
+                const appearDate = planningItemDate
+                  .clone()
+                  .subtract(timestamp.get('delayValue'), delayUnit);
+
+                return date >= appearDate;
+              } else {
```

Comment:
hummm ... ok I see.  There's a (default) default within the setting screen, but you're right you need to visit that screen.  In which case I have a question ...
(and this might be a lack of react knowledge) but I see that  src/actions/base   has set methods for each of the settings but I don't see any getters and assume its defaulting to normal behaviour (ie. just the value).  Would it be appropriate to create getters here for the default settings rather than hard code within the UI later ?

Reply:
Yeah, that default is only on the settings screen. 

> Would it be appropriate to create getters here for the default settings rather than hard code within the UI later ?

In a redux application, the "actions" files only have methods for performing actions or setting values. The equivalent of a "getter" is just accessing something on `state` (which is essentially global to a redux application). 

So I think the right approach here is one of the following: 

1) Add default values to the `AgendaDay` component just like you did on the `Settings` component (i.e. `state.base.get('agendaDefaultDeadlineDelayValue') || 5`)
2) Use the default values earlier in the app boot process, such that by the time either the `AgendaDay` or `Settings` component are loaded we have a guarantee that those fields have values. 

I think (2) is the *more* correct way to do things, but for now I'd vote for option (1). Let me know if I can clarify those any further.

Reply:
I think 2 might be nice because I'm sure there will be lots of new settings going forwards... Otherwise we need to set default default values in multiple places for each ui component.
If we create all the default values early in the boot process then they are all in the same place...
Would you like me to try something in this regard?

Reply:
Although to finish this current PR I think you're right... I'll simply add to the agenda day